addons/stock/models/stock_move.py

```diff
@@ -1837,6 +1837,9 @@ def _set_quantities_to_reservation(self):
                     continue
                 move_line.qty_done = move_line.product_uom_qty
 
+    def _clear_quantities_to_zero(self):
+        self.filtered(lambda m: m.state in ('partially_available', 'assigned')).move_line_ids.qty_done = 0
```

Comment:
Why call a `filtered` in the picking's moves and then a another filter on the method itself ?
Isn't simpler and better to do a single filter in the `_clear_quantities_to_zero` ?

Reply:
Yeah, it can be done in that way. It's just think `_clear_quantities_to_zero` maybe possible to be used in somewhere else in the future, so I want to move the unrelated part out of this. But maybe it's too early to do that.

Reply:
I just noticed I also made some change to `action_set_quantities_to_reservation`. since `_set_quantities_to_reservation` is used in other places, I think it's better to make the change in picking. And for `_clear_quantities_to_zero`, I guess it's better to keep the related functional logic together.